### PR TITLE
Incorporate unique function complexity to rank branch blockers

### PR DESCRIPTION
**Pull request by Navidem**
Similar to the logic in https://github.com/ossf/fuzz-introspector/pull/336#issue-1272902626, this PR goes one step further by differentiating uniquely reachable functions by each branch side.
For example in the following control flow consisting basic blocks A, B, C, D, and function calls to u, v, w, x, y; assume the edge A -> C is blocking, here we collect the complexity of functions `v, w` separate from `v, w, x, y` and mark if they are covered or not-covered. 
Then the branch blockers first are ranked based on `[unique_not_covered_funcs_complexity, unique_reachable_funcs_complexity]` and then the rest of reachable/covered functions compelxity.

```mermaid
graph TD;
    A-->B:u;
    A-.->C:v,w;
    B:u -->D:x,y;
    C:v,w -.->D:x,y;
```

Signed-off-by: Navid Emamdoost <navid.emamdoost@gmail.com>

### CODE REVIEW

File: src/fuzz_introspector/analysis.py [hunk 1]
```diff
@@ -446,17 +460,27 @@ def detect_branch_level_blockers(
...
+            blocked_side = bp.BranchSide.TRUE
+            blocked_unique_not_covered_com = (
+                llvm_branch.branch_true_side_unique_not_covered_complexity)
+            blocked_unique_reachable_com = llvm_branch.branch_true_side_unique_reachable_complexity
+            blocked_reachable_com = llvm_branch.branch_true_side_reachable_complexity
+            blocked_not_covered_com = llvm_branch.branch_true_side_not_covered_complexity
             side_line = llvm_branch.branch_true_side_pos
             side_line_number = side_line.split(':')[1].split(',')[0]
+            blocked_unique_funcs = list(
+                llvm_branch.get_side_unique_reachable_funcnames(bp.BranchSide.TRUE))
```
Comment (oliverchang) [inline src/fuzz_introspector/analysis.py:472]: nit: just pass `blocked_side` here?

File: src/fuzz_introspector/analysis.py [hunk 2]
```diff
@@ -446,17 +460,27 @@ def detect_branch_level_blockers(
...
+            blocked_side = bp.BranchSide.FALSE
+            blocked_unique_not_covered_com = (
+                llvm_branch.branch_false_side_unique_not_covered_complexity)
+            blocked_unique_reachable_com = llvm_branch.branch_false_side_unique_reachable_complexity
+            blocked_reachable_com = llvm_branch.branch_false_side_reachable_complexity
+            blocked_not_covered_com = llvm_branch.branch_false_side_not_covered_complexity
             side_line = llvm_branch.branch_false_side_pos
             side_line_number = side_line.split(':')[1].split(',')[0]
+            blocked_unique_funcs = list(
+                llvm_branch.get_side_unique_reachable_funcnames(bp.BranchSide.FALSE))
```
Comment (oliverchang) [inline src/fuzz_introspector/analysis.py:483]: nit: just pass `blocked_side` here?

**Review by oliverchang — APPROVED** [review]

File: src/fuzz_introspector/analysis.py (hunk 1 above)
Comment (Navidem) [reply src/fuzz_introspector/analysis.py:472]: Good point! Thanks!

File: src/fuzz_introspector/analysis.py (hunk 2 above)
Comment (Navidem) [reply src/fuzz_introspector/analysis.py:483]: Good point! Thanks!

**Pull request merged**